dropgz/pkg/embed/payload.go

```diff
@@ -34,36 +35,62 @@ func Contents() ([]string, error) {
 		if d.IsDir() {
 			return nil
 		}
-		contents = append(contents, strings.TrimPrefix(path, cwd+string(filepath.Separator)))
+		contents = append(contents, strings.TrimPrefix(path, pathPrefix))
 		return nil
 	}); err != nil {
 		return nil, errors.Wrap(err, "error walking gzfs")
 	}
 	return contents, nil
 }
 
-func Extract(path string) (io.ReadCloser, io.Closer, error) {
+// gzipCompoundReadCloser is a wrapper around the source file handle and
+// the gzip Reader on the file to provide a single Close implementation
+// which cleans up both.
+// We have to explicitly track and close the underlying Reader, because
+// the gzip readercloser# does not.
+type gzipCompoundReadCloser struct {
+	file     io.Closer
+	gzreader io.ReadCloser
+}
+
+func (rc *gzipCompoundReadCloser) Read(p []byte) (n int, err error) {
+	return rc.gzreader.Read(p)
+}
+
+func (rc *gzipCompoundReadCloser) Close() error {
+	if err := rc.gzreader.Close(); err != nil {
+		return err
+	}
+	if err := rc.file.Close(); err != nil {
```

Comment:
do we still need to close the file if gzreader.Close() fails?

Reply:
I guess we don't really need to close either because these are handles to the `embed.FS` and the process is about to exit. But maybe. I need to check the conditions that the gzreader might error on close to see if that would preclude that the underlying file be closed.

Reply:
the impl of `Close()` on the `embed.FS` file is noop 😅